stripe/src/main/java/com/stripe/android/view/StripeEditText.java

```diff
@@ -0,0 +1,172 @@
+package com.stripe.android.view;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.Color;
+import android.os.Build;
+import android.support.annotation.ColorInt;
+import android.support.annotation.ColorRes;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.inputmethod.EditorInfo;
+import android.view.inputmethod.InputConnection;
+import android.view.inputmethod.InputConnectionWrapper;
+import android.widget.EditText;
+
+import com.stripe.android.R;
+
+/**
+ * Extension of {@link EditText} that listens for users pressing the delete key when there is
+ * no text present. Google has actually made this
+ * <a href="https://code.google.com/p/android/issues/detail?id=42904">somewhat difficult</a>,
+ * but we listen here for hardware key presses, older Android soft keyboard delete presses,
+ * and modern Google Keyboard delete key presses.
+ */
+public class StripeEditText extends EditText {
+
+    @Nullable private DeleteEmptyListener mDeleteEmptyListener;
+    @Nullable private ColorStateList mCachedColorStateList;
+    private boolean mShouldShowError;
+    @ColorRes private int mColorResId;
+
+    public StripeEditText(Context context) {
+        super(context);
+        initView();
+    }
+
+    public StripeEditText(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView();
+    }
+
+    public StripeEditText(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView();
+    }
+
+    @Override
+    public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
+        return new SoftDeleteInputConnection(super.onCreateInputConnection(outAttrs), true);
+    }
+
+    /**
+     * Sets a listener that can react to the user attempting to delete the empty string.
+     *
+     * @param deleteEmptyListener the {@link DeleteEmptyListener} to attach to this view
+     */
+    public void setDeleteEmptyListener(DeleteEmptyListener deleteEmptyListener) {
+        mDeleteEmptyListener = deleteEmptyListener;
+    }
+
+    /**
+     * Sets whether or not the text should be put into "error mode," which displays
+     * the text in an error color determined by the original text color.
+     *
+     * @param shouldShowError whether or not we should display text in an error state.
+     */
+    @SuppressWarnings("deprecation")
+    public void setShouldShowError(boolean shouldShowError) {
+        if (!mShouldShowError && shouldShowError) {
```

Comment:
Some comments here would be helpful. Since we have a local variable and an instance variable with similar names, its not readily apparent what the difference between the `if` and `else if` is.